django/db/models/sql/compiler.py

```diff
@@ -999,14 +999,20 @@ def find_ordering_name(
             transform_function,
         ) = self._setup_joins(pieces, opts, alias)
 
+        # The transform function will be partially applied with the name of the first
+        # lookup if there are lookups.
+        has_lookups = "name" in getattr(transform_function, "keywords", {})
+
         # If we get to this point and the field is a relation to another model,
         # append the default ordering for that model unless it is the pk
-        # shortcut or the attribute name of the field that is specified.
+        # shortcut or the attribute name of the field that is specified or
+        # there are lookups to process.
         if (
             field.is_relation
             and opts.ordering
             and getattr(field, "attname", None) != pieces[-1]
             and name != "pk"
+            and not has_lookups
```

Comment:
What do you think about assigning `has_lookups` to the `transform_function` and be more explicit here?
```suggestion
            and not getattr(transform_function, "has_lookups", False)
```
and
```diff
diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index 7e43ec254d..58ccb8f923 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1820,6 +1820,7 @@ class Query(BaseExpression):
             final_transformer = functools.partial(
                 transform, name=name, previous=final_transformer
             )
+            final_transformer.has_lookups = True
         # Then, add the path to the query's joins. Note that we can't trim
         # joins at this stage - we will need the information about join type
         # of the trimmed joins.
```

Reply:
@felixxm Yep that's a good way to do it too - 1 function is created per invocation so it would work 👍. Explicit, readable & minimal api impact. The one con is that it could be surprising to future maintainers if function attributes aren't commonly done like this in this community? 🤔

If you're happy to do this I can squash everything down - or if you wanted to do it 😊🤷‍♂️

Reply:
As far as I'm aware, assigning variables to functions is not sth unusual :thinking: 

> If you're happy to do this I can squash everything down - or if you wanted to do it

Please do. Also, `has_transforms` is probably more accurate than `has_lookups`.